File: shared/src/types/report.ts
export enum ReportType {
  LocationDoesNotExist = 0,
  MisleadingInformation = 1,
  InappropriateContent = 2,
  Duplicate = 3,
}

export interface ReportPostDTO {
  userId: number;
  locationId: number;
  type: ReportType;
}

export interface ReportResponseDTO {
  id: number;
  userId: number;
  locationId: number;
  type: ReportType;
  createdAt: string;
}

File: backend/PulseMap/Domain/Report.cs
namespace PulseMap.Domain;

public class Report
{
    public int Id { get; set; }
    public required int UserId { get; set; }
    public User? User { get; set; }
    public required int LocationId { get; set; }
    public Location? Location { get; set; }
    public required ReportType Type { get; set; }
    public required DateTime CreatedAt { get; set; }
}

public enum ReportType
{
    LocationDoesNotExist = 0,
    MisleadingInformation = 1,
    InappropriateContent = 2,
    Duplicate = 3
}
